tosfs/core.py

```diff
@@ -203,6 +203,59 @@ def info(
 
         return self._try_dir_info(bucket, key, path, fullpath)
 
+    def rmdir(self, path: str) -> None:
+        """Remove a directory if it is empty.
+
+        Parameters
+        ----------
+        path : str
+            The path of the directory to remove. The path should be in the format
+            `tos://bucket_name/directory_path`.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the directory does not exist.
+        NotADirectoryError
+            If the path is not a directory.
+        TosfsError
+            If the directory is not empty,
+             or the path is a bucket.
+
+        Examples
+        --------
+        >>> fs = TosFileSystem()
+        >>> fs.rmdir("tos://mybucket/mydir/")
+
+        """
+        path = self._strip_protocol(path).rstrip("/") + "/"
+        bucket, key, _ = self._split_path(path)
+        if not key:
+            logger.error("Cannot remove a bucket using rmdir api.")
```

Comment:
Do we still need to add this logger, since we've already raise the TosFsError to the upper layer ?  I mean the upper layer will just get the error, and decide to print log or raise .